custom_components/rrd/__init__.py

```diff
@@ -123,47 +132,114 @@ def setup(hass, config):
             _LOGGER.error(exc)
             return False
 
-    def rrd_update(event):
-        state = event.data.get("new_state")
-        if state is None:
-            return
 
-        if state.entity_id not in entities:
+    # List of scheduled updates. Used for cancellation during the HASS shuting down.
+    cancel_callbacks = {}
+
+
+    def schedule_next_update(database):
+        # Do not schedule any new update, when HASS is shutting down
+        if (hassIsShuttingDown):
             return
 
-        _last_changed_ts = int(datetime.timestamp(state.last_changed))
-        _state = int(state_helper.state_as_number(state))
+        # Scheduling
+        step = convert_to_seconds(database[CONF_STEP])
+
+        now = time.time()
```

Comment:
Function `dt_util.utcnow()` returns `datetime`, not timestamp. So equivalent variant is `dt_util.utlnow().timestamp()`.
But [documentation ](https://docs.python.org/3/library/datetime.html) says: "Return` POSIX timestamp corresponding to the datetime instance. The return value is a float similar to that returned by time.time().". So question is, why so complicated variant? Why to use internal helper of HASS in case, when native Python lib provides same functions?

This code runs on my HASS instance, so I is tested.

Reply:
Daylight saving adjustments...

Reply:
Timestamp number is continuously growing constantly and independently. When daylight saving adjustments happen in local time, therefore the timestamp is just incremented by `+1` only. Nothing is visible in timestampt during daylight time adjustment.

Example: 1585443599 is one second before daylight saving adjustment in my Prague timezone.
```
print(datetime.fromtimestamp(1585443599))
print(datetime.fromtimestamp(1585443599 + 1))
```
displays:
```
2020-03-29 01:59:59
2020-03-29 03:00:00
```
... And it is correct. It is example, which proves a timestampt is not jumping because of daylight saving.

So I thing is it most correct to use local timestamp received by `time.time()`.

Reply:
You are right, you are using seconds only and that is sorted